app/uitls/feature_count.py

```diff
@@ -0,0 +1,16 @@
+def feature_count(knightsConfig, knight_2):
+    result = knightsConfig[knight_2]
+    result["protection"] = sum(a["protection"] for a in result["armour"])
+    result["power"] += result["weapon"]["power"]
+
+    if result["potion"] is not None:
+        if "power" in result["potion"]["effect"]:
+            result["power"] += result["potion"]["effect"]["power"]
+
+        if "protection" in result["potion"]["effect"]:
+            result["protection"] += result["potion"]["effect"]["protection"]
+
+        if "hp" in result["potion"]["effect"]:
+            result["hp"] += result["potion"]["effect"]["hp"]
+
```

Comment:
These code duplications can be also fixed, using stats = ("protection", "power", "hp") and looping on stats
Ex:
`stats = ("protection", "power", "hp")`

```
if knight["potion"] is not None:

                for statistic in stats:
                    if statistic in knight["potion"]["effect"]:
                        knight[statistic] += knight["potion"]["effect"][
                            statistic
                        ]
```

Reply:
Wow, cool, thanks)